test/test_core.jl

```diff
@@ -218,5 +248,6 @@ end
     l = @lens _.x
     @test get(l,t) === 1
 
+    # do we want this to throw an error?
     @test_throws ArgumentError (@set t.z = 3)
```

Comment:
I think it could be useful in some situations but you can already use `(t..., new_field=1)` for a named tuple `t`.  Maybe when adding a field to the deeply nested named tuple?  But then there is `modify`. It also breaks (?) the lens law `set(lens, obj, get(lens, obj)) == obj`.

If you don't need it immediately, `throw`ing for now sounds like a good choice to me.

Reply:
I had exactly the same thought (might be useful but can be done with `(t..., new_field=1)`.